test/lib/cacheWrapperTest.js

```diff
@@ -1,128 +1,465 @@
+              expect(redisWrapperStub).to.be.calledOnce();
+            });
+        });
+      });
+    });
+  });
+
+  describe('_retrieve', function() {
+    var options = {
+      segment: 'test',
+      key: 'foo'
+    };
+
+    describe('when no policy is specified', function() {
+      it('should reject the promise', function() {
+        cache._retrieve(deferred, options);
+        expect(deferred.reject).to.be.calledWith(sinon.match.typeOf('error'));
+      });
+    });
+
+    describe('when a policy is specified', function() {
+
+      describe('and the cache get is successful', function() {
+        beforeEach(function() {
+          getStub = sandbox.stub().yields(null, 'RESULT');
+          cache.__set__('policies', {
+            test: {
+              get: getStub
+            }
+          });
+          cache._retrieve(deferred, options);
+        });
+
+        it('should call the stub with the passed key argument', function() {
+          expect(getStub).to.be.calledWith('foo');
+        });
+        it('should resolve the promise with the stubbed cache result', function() {
+          expect(deferred.resolve).to.be.called();
+        });
+      });
+
+      describe('and the cache get fails', function() {
+        beforeEach(function() {
+          getStub = sandbox.stub().yields('ERROR');
+          cache.__set__('policies', {
+            test: {
+              get: getStub
+            }
+          });
+          cache._retrieve(deferred, options);
+
+        });
+        it('should call the stub with the passed key argument', function() {
+          expect(getStub).to.be.calledWith('foo');
+        });
+        it('should reject the promise with the stubbed cache result', function() {
+          expect(deferred.reject).to.be.calledWith(sinon.match.typeOf('error'));
+        });
+      });
+
+    });
+
+  });
+
+  describe('_stash', function() {
+
+    describe('when no policy is specified', function() {
+      it('should reject the promise', function() {
+        cache._stash(deferred, {});
+        expect(deferred.reject).to.be.calledWith(sinon.match.typeOf('error'));
+      });
+    });
+
+    describe('when a policy is specified', function() {
+      var stashOptions = {
+        segment: 'test',
+        key: 'foo',
+        value: 'bar'
+      };
+      describe('and the cache set is successful', function() {
+        beforeEach(function() {
+          getStub = sandbox.stub().yields(null, 'RESULT');
+          cache.__set__('policies', {
+            test: {
+              set: getStub
+            }
+          });
+          cache._stash(deferred, stashOptions);
+        });
+
+        it('should call the stub with the passed key and value arguments', function() {
+          expect(getStub).to.be.calledWith('foo', 'bar');
+        });
+        it('should resolve the promise with the stubbed cache result', function() {
+          expect(deferred.reject.callCount).to.equal(0);
```

Comment:
I think we can use `to.not.have.been.called()` here... not sure if it's easier to read